api-example.xml

```diff
@@ -0,0 +1,129 @@
+          return 'The angle of the Frictional Force is incorrect. Your angle: %.1f' % friction.angle
+
+  def check_starts_at_center_of_mass(answer, tolerance=0.25):
+      cm = answer['points']['cm']
+      for vec in answer['vectors'].values():
+          dist = math.hypot(cm.x - vec.tail.x, cm.y - vec.tail.y)
+          if dist > tolerance:
+              return 'One or more of your vectors do not start from the center of mass.'
+
+  checks = [
+      check_all_vectors_present,
+      check_compare_normal_and_gravity_lengths,
+      check_normal_angle,
+      check_gravity_angle,
+      check_friction_angle,
+      check_starts_at_center_of_mass,
+  ]
+  success_message = 'Good job!'
+
+
+  ### Python API - Do not edit after this line ####
+
+  class Point(object):
+      def __init__(self, x, y):
+          self.x = x
+          self.y = y
+
+  class Vector(object):
+      def __init__(self, x1, y1, x2, y2):
+          self.tail = Point(x1, y1)
+          self.tip = Point(x2, y2)
+          self.length = math.hypot(x2 - x1, y2 - y1)
+          angle = math.degrees(math.atan2(y2 - y1, x2 - x1))
+          if angle < 0:
+              angle += 360
+          self.angle = angle
+
+  class Grader(object):
+      def __init__(self, answer, checks, success_message='Test passed'):
+          self.answer = answer
+          self.checks = checks
+          self.success_message = success_message
+
+      def grade(self):
+          for check in self.checks:
+              result = check(self._convert_answer())
+              if result:
+                  return {'ok': False, 'msg': result}
+          return {'ok': True, 'msg': self.success_message}
+
+      def _convert_answer(self):
+          vectors = {}
+          for name, props in self.answer['vectors'].iteritems():
+              tail = props['tail']
+              tip = props['tip']
+              vectors[name] = Vector(tail[0], tail[1], tip[0], tip[1])
+
+          points = {}
+          for name, props in self.answer['points'].iteritems():
+              coords = props['coords']
+              points[name] = Point(coords[0], coords[1])
+
+          return {'vectors': vectors, 'points': points}
+
+  answer = json.loads(json.loads(ans)['answer'])
+  grader = Grader(answer, checks, success_message)
+
+  return grader.grade()
+]]>
+</script>
+<p>
+A car is moving in the positive x-direction and is slowing down. Draw the appropriate free-body diagram for this situation. You can create a draggable vector by clicking one of the buttons to the right of the interactive canvas.
+</p>
+<p>
+  <i>Be sure that the "tail" of each vector starts at the center of mass.</i> Also try to break this - I have error messages for most possible situations.
+</p>
+<customresponse cfn="vglcfn">
+  <jsinput gradefn="getInput"
+    get_statefn="getState"
+    set_statefn="setState"
+    initial_state='{}'
+    width="750"
+    height="450"
+    html_file="/static/jsapi.html"
```

Comment:
@mtyaka nit: html_file used with this example is named api-example.html. Might make sense to update it.